propertly.core/src/main/java/de/adito/propertly/core/api/DelegatingNode.java

```diff
@@ -243,7 +243,18 @@ public void reorder(@NotNull Comparator pComparator, @NotNull Set<Object> pAttri
       List<Runnable> onFinish = new ArrayList<>();
       firePropertyOrderWillBeChanged(onFinish::add, pAttributes);
       children.reorder(pComparator);
-      executeWriteOnDelegate(pDelegate -> pDelegate.reorder(Comparator.<IProperty>comparingInt(p -> children.indexOf(p.getDescription())), pAttributes));
+      executeWriteOnDelegate(pDelegate -> {
+        // extract indices from the current children first, to improve performance on large models
+        Map<IPropertyDescription<?, ?>, Integer> childrenIndices = new HashMap<>();
```

Comment:
Maybe set the size for the map on creation to prevent resizing

Reply:
Size of the map is set now, based on the childrenList-List